hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CertificateClient.java

```diff
@@ -126,6 +130,30 @@ X509Certificate getCertificate(String certSerialId)
    */
   Set<X509Certificate> getAllCaCerts();
 
+  /**
+   * This method serves as a convenience to solve the common problem in the
+   * code related to clients, and handle single SCM case and the data with
+   * which we provide one single cert to old clients.
+   * @return either the rootCA certificate, or the sub-CA certificate
+   *          of the node that served as the rootCA during bootstrap.
+   */
+  default String getSingleCACertificatePEM() throws SCMSecurityException {
+    return CertificateCodec.getPEMEncodedString(getSingleCACertificate());
+  }
```

Comment:
Since this is only used by `ServiceInfoProvider`, which already handles PEM encoding for the CA list, I think this method should be omitted.

Reply:
Yes, this makes testing a bit more complex though I think you are right, a simpler API for the CertificateClient interface is more important, I haven't realized this one before, thank you for catching it.

Reply:
There are a few more tests that need to be updated:

https://github.com/fapifta/hadoop-ozone/actions/runs/5459036651/jobs/9934647451#step:5:3792
https://github.com/fapifta/hadoop-ozone/actions/runs/5459036651/jobs/9934647503#step:5:3764
https://github.com/fapifta/hadoop-ozone/actions/runs/5459036651/jobs/9934647503#step:5:3810

Sorry for the trouble. :)

Reply:
Yes, I have adjusted the code to overcome these.